great_expectations/data_context/data_context/cloud_data_context.py

```diff
@@ -332,6 +330,7 @@ def _init_variables(self) -> CloudDataContextVariables:
 
         variables = CloudDataContextVariables(
             config=self._project_config,
+            config_provider=self._config_provider,
```

Comment:
`DataContextVariables` now leverage an instance of `ConfigurationProvider` now that substitution logic is a part of this class.